### ISSUE TITLE
Sometimes not all files passed to add() function are added

### ISSUE THREAD

**Issue by TheKysek** [ipfs/py-ipfs-api#85]
While running tests against newest versions of go-ipfs i noticed that "test_add_multiple_from_list" was seemingly randomly failing.

It seems like while adding multiple files using "add(['file1', 'file2'])" sometimes this call fails.

```
>>> a.add(['/home/krzysiek/Dokumenty/PyCharmProjects/py-ipfs-api/test/functional/fake_dir/fsdfgh', '/home/krzysiek/Dokumenty/PyCharmProjects/py-ipfs-api/test/functional/fake_dir/popoiopiu'])
[{'Hash': 'QmQcCtMgLVwvMQGu6mvsRYLjwqrZJcYtH4mboM9urWW9vX', 'Name': 'fsdfgh'}, {'Hash': 'QmYAhvKYu46rh5NcHzeu6Bhc7NG9SqkF9wySj2jvB74Rkv', 'Name': 'popoiopiu'}]
>>> a.add(['/home/krzysiek/Dokumenty/PyCharmProjects/py-ipfs-api/test/functional/fake_dir/fsdfgh', '/home/krzysiek/Dokumenty/PyCharmProjects/py-ipfs-api/test/functional/fake_dir/popoiopiu'])
{'Hash': 'QmQcCtMgLVwvMQGu6mvsRYLjwqrZJcYtH4mboM9urWW9vX', 'Name': 'fsdfgh'}
>>> a.add(['/home/krzysiek/Dokumenty/PyCharmProjects/py-ipfs-api/test/functional/fake_dir/fsdfgh', '/home/krzysiek/Dokumenty/PyCharmProjects/py-ipfs-api/test/functional/fake_dir/popoiopiu'])
[{'Hash': 'QmQcCtMgLVwvMQGu6mvsRYLjwqrZJcYtH4mboM9urWW9vX', 'Name': 'fsdfgh'}, {'Hash': 'QmYAhvKYu46rh5NcHzeu6Bhc7NG9SqkF9wySj2jvB74Rkv', 'Name': 'popoiopiu'}]
```
When this happens the following error is printed in go-ipfs daemon console:
```
13:46:35.014 ERROR commands/h: err: http: invalid Read on closed Body handler.go:288
```

**Comment by alexander255** [comment]
Thanks for reporting, but we unfortunately do not currently have time to look into fixing this. (Maybe in a couple of weeks.)

You, or anybody else interested in helping out, are very welcome to provide patches or further hints for this through.
Thanks for helping out by reporting either way!

*Sorry for the delayed response, this should have been sent on Thursday, but GitHub apparently told me: „**You can't comment at this time.**“ – And I didn't see. :unamused:*

**Comment by whereswaldon** [comment]
@TheKysek Nice catch! I also can't try to debug this at the moment, but we'd welcome any further hints as to where the error originates. It seems like the IPFS node is sometimes receiving an invalid HTTP body. It's possible that the error has something to do with the code that generates the multipart HTTP request body.

**Comment by alexander255** [comment]
This should be fixed by the new multipart code (9edc945088faf5df66cab1845bd2c87c5bc77ede), although I think the issue was already fixed before that.

**Comment by lidel** [comment]
For anyone googling  `http: invalid Read on closed Body` and finding this, the upstream error will be fixed in go-ipfs v0.4.19: https://github.com/ipfs/go-ipfs/issues/5168
